src/System.Console/tests/WindowAndCursorProps.cs

```diff
@@ -31,6 +31,15 @@ public static void BufferSize_GettingSameAsWindowSize()
     }
 
     [Fact]
+    [SkipOnTargetFramework(TargetFrameworkMonikers.Netcoreapp | TargetFrameworkMonikers.NetcoreUwp)]
+    public static void WindowWidth_WindowHeight_InvalidSize_net46()
+    {
+        Assert.Throws<IOException>(() => Console.WindowWidth = 0);
+        Assert.Throws<IOException>(() => Console.WindowHeight = 0);
+    }
+
+    [Fact]
+    [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
```

Comment:
Great summary @stephentoub. We should capture that in our breaking changes md file. As you state case (4) we should have a strong reason to make the change. In this case I don't know why we would change from ArgumentOutOfRangeException to IOException so I would argue we should fix this in core to match desktop. 

@tarekgh can you please file an issue to fix this to match the desktop behavior?